frontend/catalyst/pennylane_extensions.py

```diff
@@ -428,7 +438,85 @@ def circuit(x):
     >>> workflow(2.0)
     array(-3.14159265)
     """
-    return Grad(_ensure_differentiable(f), grad_params=_check_grad_params(method, h, argnum))
+    scalar_out = True
+    return Grad(
+        _ensure_differentiable(f), grad_params=_check_grad_params(method, scalar_out, h, argnum)
+    )
+
+
+def jacobian(f: DifferentiableLike, *, method=None, h=None, argnum=None):
+    """A :func:`~.qjit` compatible Jacobian transformation for PennyLane/Catalyst.
+
+    This function allows the Jacobian of a hybrid quantum-classical function
+    to be computed within the compiled program.
+
+    .. warning::
+
+        If parameter-shift or adjoint is specified, this will only be used
+        for internal _quantum_ functions. Classical components will be differentiated
+        using standard reverse-mode automatic differentiation.
+
+    .. warning::
+
+        Currently, higher-order differentiation or differentiation of non-QNode functions
+        is only supported by the finite-difference method.
+
+    .. note::
+
+        Any JAX-compatible optimization library, such as `JAXopt
+        <https://jaxopt.github.io/stable/index.html>`_, can be used
+        alongside ``grad`` for JIT-compatible variational workflows.
+        See the :doc:`/dev/quick_start` for examples.
+
+    Args:
+        f (Callable): a function or a function object to differentiate
+        method (str): The method used for differentiation, which can be any of
+                      ``["fd", "defer"]``,
+            where:
```

Comment:
```suggestion
        method (str): The method used for differentiation, which can be any of
    ``["fd", "defer"]``, where:
```

Reply:
to avoid the rendering issue here:

<img width="546" alt="image" src="https://github.com/PennyLaneAI/catalyst/assets/2959003/c592f6b5-2b05-40ad-9872-4e85a73544fc">